inc/utils/namespace.php

```diff
@@ -81,12 +81,19 @@ function get_elasticsearch_url() : string {
  *
  * @param array $query A full elasticsearch Query DSL array.
  * @param array $params URL query parameters to append to request URL.
+ * @param string $path The endpoint to query against, defaults to _search.
  * @return array|null
  */
-function query( array $query, array $params = [] ) : ?array {
+function query( array $query, array $params = [], string $path = '_search' ) : ?array {
+
+	// Sanitize path.
+	$path = trim( $path, '/' );
 
 	// Get URL.
-	$url = add_query_arg( $params, get_elasticsearch_url() . '/analytics*/_search' );
+	$url = add_query_arg( $params, get_elasticsearch_url() . '/analytics*/' . $path );
+
+	// Ensure URL is legit.
```

Comment:
In what cases is it "not legit"?

Reply:
I added this because the path portion is now a string parameter so could be anything, while incorrect values would likely just give an error this is an extra safety check.

I can remove it if you think it's really unnecessary.

Reply:
Updated the comment. It doesn't do any harm at least.

Reply:
My apologies; I guess what I meant was "can we note in the comment what this is checking for". New comment looks good 👍